src/logging.c

```diff
@@ -1252,6 +1260,8 @@ rcutils_ret_t rcutils_logging_format_message(
   }
 # define SET_STANDARD_COLOR_IN_STREAM(is_colorized, status)
 #endif
+static  char msg_buf[1024] = ""; 
+static  char output_buf[1024] = ""; 
```

Comment:
These changes shall be done in the original repo: https://github.com/ros2/rcutils

Remember that this is a fork

Reply:
I could try, but this is a change that really mostly makes sense in the context of small embedded systems. I'm not sure that the main ros2 project would see much reason for a change of this nature.

Reply:
You are changing the variable from stack to static section, right?

Reply:
Yes, that's what I'm doing. I could try submitting it to the ros2 repo. There shouldn't technically be anything /wrong/ with doing it that way even on more capable hardware that can handle the overhead.

Reply:
I tried submitting this to the ros2 repository. They said they can't accept the change that makes the buffer a global (static) variable, because that function is called by multiple threads in a ros2 instance. Does micro-ROS potentially have the same problem, or is micro-ROS guaranteed to be single-threaded with respect to callers of rcutils_logging_console_output_handler()?

Reply:
They can accept the other change (removing the dependency on isatty() for Zephyr builds), so I'll go ahead and push that to the upstream ros2 repo.

Reply:
micro-ROS potentially can be used in a threaded environment in some cases with some configuration, so we have the very same problem.

Reply:
OK, then I think the only safe way to do it is dynamic heap allocation (malloc). Could possibly also use locking to prevent more than one thread from calling the function at a time. Do you have any idea if this is a Zephyr-specific issue?

Reply:
For now, I think I'll leave the buffer allocation changes out of my PR, since no one else seems to have a problem with stack allocation. We'll continue to patch for our own purposes. Doyon have any comments on any of the other changes? It looks like they passed CI this time.